src/awkward/_connect/cuda/cuda_kernels/awkward_UnionArray_regular_index_getsize.cu

```diff
@@ -18,6 +20,6 @@ awkward_UnionArray_regular_index_getsize(
     uint64_t invocation_index,
     uint64_t* err_code) {
   if (err_code[0] == NO_ERROR) {
-    *size = length > 0 && *size > 0 ? *size + 1 : 1;
+    *size = *size + 1;
```

Comment:
I checked this against [the specification](https://awkward-array.org/doc/main/reference/generated/kernels.html#awkward-unionarray-regular-index-getsize) to verify that there's no condition on whether `*size` should be incremented. It looks good!

Reply:
This kernel was causing illegal memory access error for some edge cases, so I just fixed that.